python-test/features/steps/agentsProvider.py

```diff
@@ -0,0 +1,54 @@
+from behave import given, when, then
+import time
+import functions
+import os
+import docker
+
+email = os.getenv('EMAIL')
+password = os.getenv('PASSWORD')
+
+agentname = os.getenv('AGENT_NAME')
+agentTagKey = os.getenv('TAG_KEY', 'test')
+agentTagValeu = os.getenv('TAG_VALUE', 'true')
+
+
+@given('A valid authentication')
+def get_auth(context):
+    context.token = functions.generate_token(email, password)
+    assert len(context.token) > 0
```

Comment:
We should use assertion libraries that provide useful functionalities like nice error messages, comparisons beyond boolean, and improved readability.